Content/Items/Vitric/Weapons.PhoenixStorm.cs

```diff
@@ -0,0 +1,387 @@
+			(proj.ModProjectile as PhoenixStormMinion).syncItem = Item;
+			return false;
+		}
+
+		public override void AddRecipes()
+		{
+			Recipe recipe = CreateRecipe();
+			recipe.AddIngredient(ItemID.Book, 1);
+			recipe.AddIngredient(ItemID.HellstoneBar, 10);
+			recipe.AddIngredient(ModContent.ItemType<MagmaCore>(), 1);
+			recipe.AddTile(TileID.Anvils);
+			recipe.Register();
+		}
+	}
+
+	public class PhoenixStormMinion : ModProjectile
+	{
+		private const int AFTER_IMAGE_LENGTH = 6;
+
+		public Item syncItem;
+		public int swoopDelay = -1;
+
+		private float circleCounter = 0;
+		private bool swooping = false;
+		private float swoopSpeed = 0.1f;
+
+		private List<NPC> alreadyHit = new List<NPC>();
+
+		private List<Vector2> oldPos = new List<Vector2>();
+		private List<float> oldRot = new List<float>();
+
+		private List<Vector2> swoopTrajectory = new List<Vector2>();
+
+		private NPC target;
+
+		private Trail trail;
+		private List<Vector2> cache;
+
+		private float glowOpacity = 1;
+
+		private Player player => Main.player[Projectile.owner];
+
+		public override string Texture => AssetDirectory.VitricItem + Name;
+
+		public override void SetStaticDefaults()
+		{
+			DisplayName.SetDefault("Phoenix");
+			Main.projFrames[Projectile.type] = 7;
+
+			Main.projPet[Projectile.type] = true; // Denotes that this Projectile is a pet or minion
+			ProjectileID.Sets.MinionTargettingFeature[Projectile.type] = true; // This is necessary for right-click targeting
+			ProjectileID.Sets.MinionSacrificable[Projectile.type] = true; // This is needed so your minion can properly spawn when summoned and replaced when other minions are summoned			 
+			ProjectileID.Sets.CultistIsResistantTo[Projectile.type] = true; // Don't mistake this with "if this is true, then it will automatically home". It is just for damage reduction for certain NPCs
+		}
+
+		public sealed override void SetDefaults()
+		{
+			Projectile.width = 32;
+			Projectile.height = 32;
+
+			Projectile.tileCollide = false;
+
+			Projectile.friendly = true;
+
+			Projectile.minion = true;
+
+			Projectile.minionSlots = 0.5f;
+
+			Projectile.penetrate = 1;
+			swoopSpeed = Main.rand.NextFloat(0.025f, 0.055f);
+		}
+
+		public override bool? CanCutTiles()
+		{
+			return false;
+		}
+
+		public override void AI()
+		{
+			#region Active check
+			if (player.dead || !player.active) // This is the "active check", makes sure the minion is alive while the Player is alive, and despawns if not
+				player.ClearBuff(BuffType<PhoenixStormSummonBuff>());
+
+			if (player.HasBuff(BuffType<PhoenixStormSummonBuff>()))
+				Projectile.timeLeft = 2;
+			#endregion
+
+			#region Find Target
+			if (player.HasMinionAttackTargetNPC)
+				target = Main.npc[player.MinionAttackTargetNPC];
+			else
+				target = Main.npc.Where(n => n.active && n.Distance(player.Center) < 800 && n.CanBeChasedBy(Projectile)).OrderBy(n => n.Distance(player.Center)).FirstOrDefault();
+
+			#endregion
+
+			#region trail stuff
```

Comment:
Is there a good reason for these regions here? Regions this small tend to do nothing but hurt readability.

Reply:
If you want to annotate the AI regular comments will do just fine